firefox-ios/Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -477,24 +477,6 @@ extension TabScrollingController: UIScrollViewDelegate {
             setOffset(y: 0, for: scrollView)
         }
 
-        let toolbarState = store.state.screenState(Client.ToolbarState.self, for: .toolbar, window: windowUUID)
-
-        guard let toolbarState,
-              let borderPosition = toolbarState.addressToolbar.borderPosition
-        else { return }
-
-        // Only dispatch the action to update the toolbar border if needed, which is only if either
-        // a) we scroll down, and the toolbar border is not already at the bottom (so we show it), or
-        // b) we scroll up past the top of the scroll view, and the border is currently at the bottom (so we hide it)
-        if (scrollView.contentOffset.y > 0 && borderPosition != .bottom)
-           || (scrollView.contentOffset.y < 0 && borderPosition != .none) {
-            store.dispatch(
-                GeneralBrowserMiddlewareAction(
-                    scrollOffset: scrollView.contentOffset,
-                    windowUUID: windowUUID,
-                    actionType: GeneralBrowserMiddlewareActionType.websiteDidScroll))
```

Comment:
for context, are we removing this because it is already handled in the Home Page and making this redundant?

Reply:
This code was specifically for showing/hiding the border when scrolling on webpages, which we don't care about because we always want to show the address bar border when on a webpage, and that is being handled in the new changes to `AddressBarState`. 

Scrolling on the homepage is handled in [HomepageViewController](https://github.com/mozilla-mobile/firefox-ios/blob/8df05c435768635deac7de696ba61cb061721cd1/firefox-ios/Client/Frontend/Home/HomepageViewController.swift#L432-L439).

Reply:
Oh okay, thanks for the clarity, didn't realize the design requirements for this. thanks!

Reply:
Ahh, I was totally wrong on this - as @thatswinnie pointed out in a [comment below](https://github.com/mozilla-mobile/firefox-ios/pull/22120#discussion_r1774793729), the design requirement is to show the top address bar bottom border on scroll - so I will be adding this back 😅

Reply:
Oh okay cool, thank you! Feel free to ping me if you still like a review from me. thanks!